registry/store/external_registry.go

```diff
@@ -27,15 +27,24 @@ type ExternalRegistry interface {
 	ListServiceInstancesByName(namespace auth.Namespace, name string) (map[string]*ServiceInstance, error)
 	ListAllServiceInstances(namespace auth.Namespace) (map[string]ServiceInstanceMap, error)
 	InsertServiceInstance(namespace auth.Namespace, instance *ServiceInstance) error
-	DeleteServiceInstance(key string) (int, error)
+	DeleteServiceInstance(namespace auth.Namespace, instanceID string) (int, error)
 }
 
+const (
+	registryRecordKey = "reg"
+	namespaceKey      = "ns"
+	instanceKey       = "inst"
+	keySeparator      = ":"
+)
+
 // DBKey represents the service instance key
 type DBKey struct {
 	Namespace  string
 	InstanceID string
 }
 
+// Database key format:
+// reg:inst:<instance id>:ns:<namespace>
 func (dbk *DBKey) String() string {
-	return fmt.Sprintf("%s%s%s", dbk.InstanceID, keySeparator, dbk.Namespace)
+	return fmt.Sprintf("%s%s%s%s%s%s%s%s%s", registryRecordKey, keySeparator, instanceKey, keySeparator, dbk.InstanceID, keySeparator, namespaceKey, keySeparator, dbk.Namespace)
```

Comment:
Also, given that there's no good reason otherwise, I would suggest to order the key elements as follows:
`reg:ns:<namespace>:inst:<instance id>`.

This should already improve the performance of regex matching during scans, and could potentially improve even fruther in the future, in case redis will support optimized scans over sorted key space.

Also, it feels more natural and hierarchical this way.